gcloud-java-dns/src/main/java/com/google/gcloud/dns/ManagedZoneInfo.java

```diff
@@ -0,0 +1,334 @@
+
+  private static final long serialVersionUID = 201601191647L;
+  private final String name;
+  private final Long id;
+  private final Long creationTimeMillis;
+  private final String dnsName;
+  private final String description;
+  private final String nameServerSet;
+  private final List<String> nameServers;
+
+  /**
+   * A builder for {@code ManagedZoneInfo}.
+   */
+  public static class Builder {
+    private String name;
+    private Long id;
+    private Long creationTimeMillis;
+    private String dnsName;
+    private String description;
+    private String nameServerSet;
+    private List<String> nameServers = new LinkedList<>();
+
+    private Builder() {
+    }
+
+    private Builder(Long id) {
+      this.id = checkNotNull(id);
+    }
+
+    private Builder(String name) {
+      this.name = checkNotNull(name);
+    }
+
+    private Builder(String name, Long id) {
+      this.name = checkNotNull(name);
+      this.id = checkNotNull(id);
+    }
+
+    /**
+     * Creates a builder from an existing ManagedZoneInfo object.
+     */
+    Builder(ManagedZoneInfo info) {
+      this.name = info.name;
+      this.id = info.id;
+      this.creationTimeMillis = info.creationTimeMillis;
+      this.dnsName = info.dnsName;
+      this.description = info.description;
+      this.nameServerSet = info.nameServerSet;
+      this.nameServers.addAll(info.nameServers);
+    }
+
+    /**
+     * Sets a mandatory user-provided name for the zone. It must be unique within the project.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets an id for the managed zone which is assigned to the managed zone by the server.
+     */
+    public Builder id(long id) {
+      this.id = id;
+      return this;
+    }
+
+    /**
+     * Sets the time when this managed zone was created.
+     */
+    Builder creationTimeMillis(long creationTimeMillis) {
+      checkArgument(creationTimeMillis >= 0, "The timestamp cannot be negative.");
+      this.creationTimeMillis = creationTimeMillis;
+      return this;
+    }
+
+    /**
+     * Sets a mandatory DNS name of this managed zone, for instance "example.com.".
+     */
+    public Builder dnsName(String dnsName) {
+      this.dnsName = checkNotNull(dnsName);
+      return this;
+    }
+
+    /**
+     * Sets a mandatory description for this managed zone. The value is a string of at most 1024
+     * characters (this limit is posed by Google Cloud DNS; gcloud-java does not enforce the limit)
+     * which has no effect on the managed zone's function.
+     */
+    public Builder description(String description) {
+      this.description = checkNotNull(description);
+      return this;
+    }
+
+    /**
+     * Optionally specifies the NameServerSet for this managed zone. A NameServerSet is a set of DNS
+     * name servers that all host the same ManagedZones.
+     */
+    public Builder nameServerSet(String nameServerSet) {
```

Comment:
Acknowledged. This is a pending question. Both gcloud-node and gcloud-ruby handle it as a string without parsing. I will keep asking and dig through some code.

Documentation fixed.